fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -466,6 +473,113 @@ where
     }
 }
 
+impl<S, Tx, Ecal> Interpreter<S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    fn upgrade_inner(
+        upgrade: &mut Upgrade,
+        storage: &mut S,
+        initial_balances: InitialBalances,
+        gas_costs: &GasCosts,
+        fee_params: &FeeParameters,
+        base_asset_id: &AssetId,
+        gas_price: Word,
+    ) -> Result<(), InterpreterError<S::DataError>> {
+        let metadata = upgrade.metadata().as_ref();
+        debug_assert!(
+            metadata.is_some(),
+            "`upgrade_inner` is called without cached metadata"
+        );
```

Comment:
Can't we compute the metadata here?

Reply:
We can, but it was calculated before in the `Checked` transaction. It is just a debug assert for the future if someone decides to change the code or behavior later.

Recalculation of it breaks the idea of the metadata=D